sdk/nodejs/runtime/invoke.ts

```diff
@@ -216,8 +216,14 @@ function createInvokeRequest(tok: string, serialized: any, provider: string | un
     return req;
 }
 
-function getProvider(tok: string, opts: InvokeOptions) {
-    return opts.provider ? opts.provider : opts.parent ? opts.parent.getProvider(tok) : undefined;
+async function getProvider(tok: string, opts: InvokeOptions): Promise<ProviderResource|undefined> {
+    if (opts.provider) {
+        return opts.provider;
+    }
+    if (opts.parent) {
+        return await output(opts.parent.getProviderInput(tok)).promise(/*withUnknowns*/false);
```

Comment:
See this is why I'm not keen on this "allow Output<Resource> everywhere we allow Resource". You can't send `unknown` to the engine for `provider`, but if you leave it blank that tells the resource to use the default provider. I think the only _correct_ thing to do if we allow Outputs in these locations is to behave the same as if this whole resource was inside an Apply, that is we skip the entire registerResourceCall.